zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -386,61 +391,83 @@ impl DiskDb {
+        // Transparent
+        "balance_by_transparent_addr",
+        "tx_loc_by_transparent_addr_loc",
+        "utxo_by_out_loc",
+        "utxo_loc_by_transparent_addr_loc",
+        // Sprout
+        "sprout_nullifiers",
+        "sprout_anchors",
+        "sprout_note_commitment_tree",
+        // Sapling
+        "sapling_nullifiers",
+        "sapling_anchors",
+        "sapling_note_commitment_tree",
+        // Orchard
+        "orchard_nullifiers",
+        "orchard_anchors",
+        "orchard_note_commitment_tree",
+        // Chain
+        "history_tree",
+        "tip_chain_value_pool",
+    ];
+
     /// Opens or creates the database at `config.path` for `network`,
     /// and returns a shared low-level database wrapper.
     pub fn new(config: &Config, network: Network) -> DiskDb {
         let path = config.db_path(network);
-        let db_options = DiskDb::options();
 
-        let column_families = vec![
-            // Blocks
-            rocksdb::ColumnFamilyDescriptor::new("hash_by_height", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("height_by_hash", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("block_header_by_height", db_options.clone()),
-            // Transactions
-            rocksdb::ColumnFamilyDescriptor::new("tx_by_loc", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("hash_by_tx_loc", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("tx_loc_by_hash", db_options.clone()),
-            // Transparent
-            rocksdb::ColumnFamilyDescriptor::new("balance_by_transparent_addr", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new(
-                "tx_loc_by_transparent_addr_loc",
-                db_options.clone(),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new("utxo_by_out_loc", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new(
-                "utxo_loc_by_transparent_addr_loc",
-                db_options.clone(),
-            ),
-            // Sprout
-            rocksdb::ColumnFamilyDescriptor::new("sprout_nullifiers", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("sprout_anchors", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("sprout_note_commitment_tree", db_options.clone()),
-            // Sapling
-            rocksdb::ColumnFamilyDescriptor::new("sapling_nullifiers", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new("sapling_anchors", db_options.clone()),
-            rocksdb::ColumnFamilyDescriptor::new(
-                "sapling_note_commitment_tree",
-                db_options.clone(),
+        let running_version = database_format_version_in_code();
+        let disk_version = database_format_version_on_disk(config, network)
+            .expect("unable to read database format version file");
+
+        match running_version.cmp(&disk_version) {
```

Comment:
There should be a "creating new database" case here, I could do it by returning Ok(None) when there's no on disk version. Currently we're logging new databases as an older version, which is incorrect.

Reply:
Fixed in e90b718d79